src/GovUk.Education.ExploreEducationStatistics.Content.Model/ReleaseFile.cs

```diff
@@ -8,15 +9,17 @@ public class ReleaseFile
         public Guid Id { get; set; }
 
         [JsonIgnore]
-        public Release Release { get; set; }
-        
+        public Release Release { get; set; } = null!;
```

Comment:
Why the need for the exclamation mark in this syntax?  If it's possible to set this to null, then should it be `public Release?` rather than `public Release`   ?

Then presumably you wouldn't have to tell the compiler to be forgiving?

Reply:
As we have enabled nullables, reference types need to be initialised upon construction, however, this really requires you to have explicit constructor methods that initialize everything correctly.  Unfortunately, this doesn't play well with Entity Framework as I _think_ it typically uses setters to construct everything, hence we need an empty constructor (hence fields are potentially null upon construction).

I believe a similar thing happens with Hibernate, so you typically need to have an empty constructor method.

So basically, a workaround for this is to use the null-forgiving operator to basically re-assure the C# compiler that everything is fine. By using it we say that our field will be initialized before we actually use it - by something like Entity Framework.

Reply:
Good stuff, thanks!